homeassistant/helpers/reload.py

```diff
@@ -156,7 +156,7 @@ async def async_integration_yaml_config(
     hass: HomeAssistant,
     integration_name: str,
     *,
-    raise_on_failure: Literal[False] | bool,
+    raise_on_failure: bool,
```

Comment:
```suggestion
    raise_on_failure: Literal[False],
```
The previous overload handles the `True` case, so this should be fine.